src/main/java/Duke.java

```diff
@@ -1,10 +1,283 @@
+                            if (curr.equals("/to")) {
+                                while (st.hasMoreTokens()) {
+                                    sbTo.append(" ").append(st.nextToken());
+                                }
+                            } else {
+                                sbFrom.append(" ").append(curr);
+                            }
+                        }
+                    } else {
+                        sb.append(" ").append(token);
+                    }
+                }
+                String name = sb.toString();
+                String from = sbFrom.toString();
+                String to = sbTo.toString();
+                if (!name.isEmpty() && !from.isEmpty() && !to.isEmpty()) {
+                    task = new Event(name, false, from, to);
+                } else {
+                    throw new ftException("Error: Please tell me your event and its from/to dates");
+                }
+                break;
+            default:
+                throw new ftException("Error: Invalid Task Type");
+        }
+        myList.add(task);
+        updateTask();
+        System.out.println("    ____________________________________________________________\n"
+                + "    Completed. I've added this task: \n    "
+                + task
+                + "\n    Now you have " + myList.size() + " tasks in the list.\n"
+                + "    ____________________________________________________________\n");
+    }
+
+    private void showList() {
+        System.out.println("    ____________________________________________________________\n"
+                + "    Here are the tasks in your list:");
+        for (int i = 1; i < myList.size() + 1; i++) {
+            Task task = myList.get(i-1);
+            if (task.isDone()) {
+                System.out.println("    " + i + "." + task);
+            } else {
+                System.out.println("    " + i + "." + task);
+            }
+        }
+        System.out.println("    ____________________________________________________________\n");
+    }
+
+    private void mark(StringTokenizer st) throws ftException {
+        if (!st.hasMoreTokens()) {
+            throw new ftException("Error: No index provided");
+        }
+        int i = Integer.parseInt(st.nextToken());
+        if ((0 < i) && (i <= myList.size())) {
+            Task task = myList.get(i - 1);
+            task.mark();
+            updateTask();
+            System.out.println("    ____________________________________________________________\n"
+                    + "    Nice! I've marked this task as done:\n"
+                    + "      " + task.toString()
+                    + "\n    ____________________________________________________________\n");
+        } else {
+            throw new ftException("Error: Please provide valid index");
+        }
+    }
+
+    private void unmark(StringTokenizer st) throws ftException {
```

Comment:
Since your other methods are in the form of `deleteTask` and `updateTask` etc, to be consistent it would be better if these were `unmarkTask` and `markTask` as well